code/modules/antagonists/_common/antag_spawner.dm

```diff
@@ -10,6 +10,69 @@
 /obj/item/antag_spawner/proc/equip_antag(mob/target)
 	return
 
+//sleeper agents for nuke ops
+/obj/item/antag_spawner/sleeper_activator
+	name = "sleeper agent activation remote"
+	desc = "A remote used to activate a deep cover sleeper agent aboard the Cyberiad, which will aid us from the inside."
+	icon = 'icons/obj/device.dmi'	//same as borg tele
+	icon_state = "locator"
+	var/checking = FALSE
+	var/TC_cost = 0
+	var/desired_objective
+
+/obj/item/antag_spawner/sleeper_activator/attack_self(mob/user)
+	if(used)
+		to_chat(user, "<span class='warning'>[src] has already been used.</span>")
+		return
+	if(!(user.mind in ticker.mode.syndicates))
+		to_chat(user, "<span class='danger'>AUTHENTICATION FAILURE. ACCESS DENIED.</span>")
+		return
+	if(checking)
+		to_chat(user, "<span class='warning'>[src] is already checking for sleeper agents to activate!</span>")
+		return
+	desired_objective = stripped_input(user, "What objective do you wish to give the sleeper agent?", "Objective")
+	checking = TRUE
+	used = TRUE
+	to_chat(user, "<span class='notice'>Activating sleeper agent, stand by.</span>")
+	var/list/agent_candidates = get_candidate_list()
+	if(!length(agent_candidates))
+		to_chat(user, "<span class='warning'>[src] failed to locate any potential sleeper agents on the station!</span>")
+		used = FALSE
+		checking = FALSE
+		return
+	var/datum/mind/agent = pick(agent_candidates)
+	ticker.mode.sleeper_agents += agent
+	agent.special_role = SPECIAL_ROLE_TRAITOR
+	ticker.mode.forge_sleeper_objectives(agent, desired_objective)	//gives our sleeper his objectives
+	ticker.mode.greet_sleeper(agent)
+	ticker.mode.equip_sleeper(agent.current)
+	agent.current.faction = list("syndicate")
+	to_chat(user, "<span class='notice'>[agent.name] the [agent.assigned_role] has been activated as a Sleeper Agent.</span>")
+	qdel(src)	//destroys itself after use
+
+
+
+
+/obj/item/antag_spawner/sleeper_activator/proc/get_candidate_list()
+	var/list/agent_candidates = list()
+	for(var/mob/living/player in GLOB.mob_list)
+		if(!player.client)
+			continue 
+		if(player.stat == DEAD)
+			continue
+		if(player.mind.special_role)
+			continue
+		if(!ishuman(player))
```

Comment:
If you typecheck for `mob/living/carbon/human` within your for loop you can remove this secondary typecheck.